native/libc.c

```diff
@@ -383,6 +386,30 @@ func_get_terminal_width(PyObject *self, PyObject *unused) {
   return PyLong_FromLong(w.ws_col);
 }
 
+static PyObject *
+func_wcswidth(PyObject *self, PyObject *args){
+    char *string;
+    if (!PyArg_ParseTuple(args, "s", &string)) {
+        return NULL;
+    }
+
+    const char *old_locale = setlocale(LC_CTYPE, NULL);
+    if (setlocale(LC_CTYPE, "en_US.UTF-8") == NULL) {
+        PyErr_SetString(PyExc_SystemError, "en_US.UTF-8 is not a valid locale");
+        return NULL;
+    }
+    int len = mbstowcs(NULL, string, 0);
+    if (len == -1) {
+        PyErr_SetString(PyExc_UnicodeError, "Invalid UTF-8 string");
```

Comment:
Maybe change this to "mbstowcs error" so we don't get confused, i.e. think it's from CPython